beluga/include/beluga/sensor/likelihood_field_model_common.hpp

```diff
@@ -0,0 +1,168 @@
+// Copyright 2022-2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_SENSOR_LIKELIHOOD_FIELD_MODEL_COMMON_HPP
+#define BELUGA_SENSOR_LIKELIHOOD_FIELD_MODEL_COMMON_HPP
+
+#include <algorithm>
+#include <beluga/actions/overlay.hpp>
+#include <beluga/algorithm/distance_map.hpp>
+#include <beluga/sensor/data/occupancy_grid.hpp>
+#include <beluga/sensor/data/value_grid.hpp>
+#include <cmath>
+#include <random>
+#include <range/v3/action/transform.hpp>
+#include <sophus/se2.hpp>
+#include <vector>
+
+/**
+ * \file
+ * \brief Implementation of a likelihood field common sensor model for range finders.
+ */
+
+namespace beluga {
+
+/// Parameters used to construct a LikelihoodFieldModelCommon instance.
+/**
+ * See Probabilistic Robotics \cite thrun2005probabilistic Chapter 6.4, particularly Table 6.3.
+ */
+struct LikelihoodFieldModelCommonParam {
+  /// Maximum distance to obstacle.
+  /**
+   * When creating a distance map, if the distance to an obstacle is higher than the value specified here,
+   * then this value will be used.
+   */
+  double max_obstacle_distance = 100.0;
+  /// Maximum range of a laser ray.
+  double max_laser_distance = 2.0;
+  /// Weight used to combine the probability of hitting an obstacle.
+  double z_hit = 0.5;
+  /// Weight used to combine the probability of random noise in perception.
+  double z_random = 0.5;
+  /// Standard deviation of a gaussian centered arounds obstacles.
+  /**
+   * Used to calculate the probability of the obstacle being hit.
+   */
+  double sigma_hit = 0.2;
+  /// Whether to model unknown space or assume it free.
+  bool model_unknown_space = false;
+};
+
+/// Likelihood field common sensor model for range finders.
+/**
+ * This model relies on a pre-computed likelihood map of the environment.
```

Comment:
@fbattocchia if we move `LikelihoodFieldModel` documentation here, we should deduplicate it there. See the `@copydoc`  command in Doxygen.